plugins/forc-postgres/src/ops/forc_postgres_createdb.rs

```diff
@@ -0,0 +1,164 @@
+            database_dir,
+            auth_method,
+            persistent,
+            timeout,
+            migration_dir,
+            postgres_version,
+            ..
+        } = val;
+        Self {
+            name,
+            user,
+            password,
+            port: port.parse::<u16>().expect("Invalid port."),
+            database_dir,
+            auth_method,
+            persistent,
+            timeout,
+            migration_dir,
+            postgres_version,
+        }
+    }
+}
+
+impl From<IndexerConfig> for PgEmbedConfig {
+    fn from(val: IndexerConfig) -> Self {
+        let IndexerConfig { database, .. } = val;
+
+        match database {
+            DatabaseConfig::Postgres {
+                user,
+                password,
+                port,
+                database: name,
+                ..
+            } => Self {
+                name,
+                user,
+                password,
+                port: port.parse::<u16>().expect("Invalid port."),
+                database_dir: None,
+                auth_method: "plain".to_string(),
+                persistent: false,
+                timeout: None,
+                migration_dir: None,
+                postgres_version: "v14".to_string(),
+            },
+        }
+    }
+}
+
+pub async fn init(command: CreateDbCommand) -> anyhow::Result<()> {
+    let CreateDbCommand {
+        name,
+        database_dir,
+        migration_dir,
+        postgres_version,
+        start,
+        config,
+        ..
+    } = command.clone();
+
+    let database_dir = database_dir_or_default(database_dir.as_ref(), &name);
+    let pgconfig: PgEmbedConfig = command.into();
+
+    let fetch_settings = PgFetchSettings {
+        version: into_postgres_version(&postgres_version),
+        ..Default::default()
+    };
+
+    let mut pg = PgEmbed::new(pgconfig.clone().into(), fetch_settings).await?;
+
+    let pg_db_uri = pg.full_db_uri(&name);
+
+    info!("Downloading, unpacking, and bootstrapping database.");
+    let pb = ProgressBar::new_spinner();
+    pb.enable_steady_tick(Duration::from_millis(120));
+    pb.set_style(
+        ProgressStyle::with_template("{spinner:.blue} {msg}")
+            .unwrap()
+            .tick_strings(&[
+                "▹▹▹▹▹",
+                "▸▹▹▹▹",
+                "▹▸▹▹▹",
+                "▹▹▸▹▹",
+                "▹▹▹▸▹",
+                "▹▹▹▹▸",
+                "▪▪▪▪▪",
+            ]),
+    );
+    pb.set_message("⏱  Setting up database...");
+
+    pg.setup().await?;
+
+    info!("🎬 Starting database at '{pg_db_uri}'.");
+
+    pg.start_db().await?;
+
+    info!("💡 Creating database at '{pg_db_uri}'.");
+
+    match pg.create_database(&name).await {
+        Ok(_) => {}
+        Err(e) => {
+            if name == DEFAULT_DATABASE {
+                info!("\nDefault database {DEFAULT_DATABASE} already exists.\n");
+            } else {
+                anyhow::bail!(e);
+            }
+        }
+    }
```

Comment:
Nit: Is it possible to use an `if let Err(e)...` here as we don't care about the `Ok` arm?